### ISSUE TITLE
Tests fail with pytest 3

### ISSUE THREAD

**Issue by ViviCoder** [astropy/astroquery#735]
Due to the use of features that are deprecated in this version of pytest, some tests fail ([see Debian bug report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=835238)).

**Comment by evertrol** [comment]
There is already [an issue filed](https://github.com/astropy/astropy/issues/5277) at the astropy issue tracker (as this is more of an astropy issue).
Are you using `ASTROPY_USE_SYSTEM_PYTEST`? Because I can only get these errors when setting that explicitly.

**Comment by ViviCoder** [comment]
The errors are not exactly the same as for astropy, but the cause is the same, indeed.
Thank you for pointing it out.
I do not explicitely use ASTROPY_USE_SYSTEM_PYTEST (is this an environment variable?), but maybe the system astropy library uses it.
I will check.
Anyway, it is part of the Debian policy to use system libraries instead of embedded ones.

**Comment by evertrol** [comment]
@ViviCoder Yes, `ASTROPY_USE_SYSTEM_PYTEST` is a environment variable. I don't know why your unit tests don't use the built-in test helper and rely on the system one. That may depend on how precisely you are running the tests. I run them through
```bash
python setup.py test
```
in the cloned source code directory. 
Perhaps you run them after installation, through e.g. the Python prompt? (though that also works fine for me.)

I expect that once astropy fixes the issue, it will be fine for astroquery as well (except that it will be longer before it shows up on pypi than on current astropy master).

**Comment by ViviCoder** [comment]
As far as I know, our tests are run using the same command line, automatically called by the packaging program.
They can also be run after installation, but this is not the case here.

In the current issue, part (if not all) of the offending code is in astroquery, not in astropy, so I guess some work will be needed here as well.
Perhaps it is a different issue, after all.

**Comment by bsipocz** [comment]
Thanks @ViviCoder for the report.

**Comment by bsipocz** [comment]
I believe the pytest3 incompatibilities were fixed in #837. Could you @ViviCoder check whether it indeed fixed this issue?
